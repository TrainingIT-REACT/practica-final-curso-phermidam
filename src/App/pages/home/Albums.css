.content{
    width: 90%;
    margin: auto;
}

.content .albums{
    text-align: center;
}

.content .albums .album{
    width: 170px;
    margin: 30px 15px;
    display: inline-block;
    cursor: pointer;
}

.content .albums .album .img{
    overflow: hidden;
}

.content .albums .album .img img{
    width: 100%;
    transition: transform 0.5s;
}

.content .albums .album:hover .img img{
    transform: scale(1.1);
    transition: transform 0.35s;
}

.content .albums .album:hover .title{
    font-weight: bold;
}