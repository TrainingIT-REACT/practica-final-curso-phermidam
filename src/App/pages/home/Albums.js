import React from "react";

import {Link} from "react-router-dom";

// Css
import './Albums.css';

class Albums extends React.Component{
    render(){
        let items = this.props.data.map((album, i) => 
            <Link to={'/album/'+album.id} className="album" key={album.id}>
                <div className="img">
                    <img src={"http://localhost:3001"+album.cover} alt="Imagen del álbum" title="Imagen del álbum" />
                </div>
                <div className="title">{album.name}<br/><i>{album.artist}</i></div>
            </Link>
        );

        return <div className="content">
            <h2>Álbumes más escuchados</h2>
            <div className="albums">{items}</div>
        </div>;
    }
}

export default Albums;