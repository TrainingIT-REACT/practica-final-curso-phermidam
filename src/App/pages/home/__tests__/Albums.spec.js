import React from 'react';
import { shallow } from 'enzyme';

import Albums from "../Albums";

describe(Albums, () => {
    describe("Render", () => {
        let wrapper;

        let data=[
            {
                id: 1,
                cover: '/images/cover.jpg',
                name: 'Prueba',
                artist: 'Artista'
            },
            {
                id: 2,
                cover: '/images/cover.jpg',
                name: 'Prueba 2',
                artist: 'Artista'
            }
        ];

        beforeEach(() => {
            wrapper = shallow(<Albums data={data}/>);
        });

        it('should add the HTML elements', () => {
            expect(wrapper.is('#albums')).toBeTruthy();
            expect(wrapper.find('.albums').length).toBe(1);
            expect(wrapper.find('.albums')).toMatchSelector(".albums");
            expect(wrapper.find('.album').length).toBe(2);
            expect(wrapper.find('.album .title').first().text()).toBe("PruebaArtista");

            expect(wrapper).toMatchSnapshot();
        });
    });
});