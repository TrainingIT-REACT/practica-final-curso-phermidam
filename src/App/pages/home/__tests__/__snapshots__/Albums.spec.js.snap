// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Render should add the HTML elements 1`] = `
<div
  className="content"
  id="albums"
>
  <h2>
    Álbumes más escuchados
  </h2>
  <div
    className="albums"
  >
    <Link
      className="album"
      key="1"
      to="/album/1"
    >
      <div
        className="img"
      >
        <img
          alt="Imagen del álbum"
          src="http://localhost:3001/images/cover.jpg"
          title="Imagen del álbum"
        />
      </div>
      <div
        className="title"
      >
        Prueba
        <br />
        <i>
          Artista
        </i>
      </div>
    </Link>
    <Link
      className="album"
      key="2"
      to="/album/2"
    >
      <div
        className="img"
      >
        <img
          alt="Imagen del álbum"
          src="http://localhost:3001/images/cover.jpg"
          title="Imagen del álbum"
        />
      </div>
      <div
        className="title"
      >
        Prueba 2
        <br />
        <i>
          Artista
        </i>
      </div>
    </Link>
  </div>
</div>
`;
